daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/FNFunctions.scala

```diff
@@ -685,7 +685,7 @@ case class FNCeiling(recipe: CompiledDPath, argType: NodeInfo.Kind) extends FNOn
 
     case NodeInfo.Decimal => {
       val bd = asBigDecimal(value)
-      bd.round(new MathContext(0, RoundingMode.CEILING))
+      bd.setScale(0, RoundingMode.CEILING)
```

Comment:
If this is a side-effect on this big decimal object, that object could be referenced from say, an element in the infoset. A formula elsewhere in the schema could be accessing its value and calling a round on it. 

So then the original element would have changed also.

I think you need to make a copy before you setScale.